src/OnboardingSPA/steps/TheFork/index.js

```diff
@@ -79,23 +91,30 @@ const TheFork = () => {
 
 		window.location.replace( pluginDashboardPage );
 	};
+
 	const content = getContents();
 	const navigate = useNavigate();
 
 	const handleMigration = () => {
 		if ( canMigrateSite ) {
-			const updates = injectMigrationStep( allSteps, stepTheFork );
-			updateAllSteps( updates.allSteps );
+			const migrationStepExists = allSteps.some(
+				( step ) => step.path === stepSiteGenMigration.path
+			);
+
+			if ( ! migrationStepExists ) {
+				const updates = injectMigrationStep( allSteps, stepTheFork );
+				updateAllSteps( updates.allSteps );
+			}
+			trackOnboardingEvent(
+				new OnboardingEvent(
+					ACTION_SITEGEN_FORK_OPTION_SELECTED,
+					'MIGRATE'
+				)
+			);
```

Comment:
In Welcome step do we have an alternate? that says Migration Initiated or clicked?

Reply:
Not as of now. ACTION_SITEGEN_FORK_OPTION_SELECTED is for the Welcome step as well, same as The Fork step